api/client/inspect.go

```diff
@@ -60,16 +62,16 @@ func (cli *DockerCli) CmdInspect(args ...string) error {
 
 	for _, name := range cmd.Args() {
 		if *inspectType == "" || *inspectType == "container" {
-			obj, _, err = readBody(cli.call("GET", "/containers/"+name+"/json?"+v.Encode(), nil, nil))
+			obj, statusCode, err = readBody(cli.call("GET", "/containers/"+name+"/json?"+v.Encode(), nil, nil))
 			if err != nil {
 				if err == errConnectionFailed {
 					return err
 				}
 				if *inspectType == "container" {
-					if strings.Contains(err.Error(), "No such") {
-						fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
-					} else {
+					if statusCode != http.StatusNotFound {
 						fmt.Fprintf(cli.err, "%s", err)
+					} else {
+						fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
```

Comment:
Just a nit; I think the if/else should be swapped, I.e. check for `statusCode == http.StatusNotFound`. It's more consistent with other if/elses in this file.